pallets/author-mapping/src/lib.rs

```diff
@@ -283,4 +283,40 @@ pub mod pallet {
 			Self::account_and_deposit_of(author_id).map(|info| info.account)
 		}
 	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_runtime_upgrade() -> Weight {
+			let pallet_prefix: &[u8] = b"AuthorMapping";
+			let old_storage_item_prefix: &[u8] = b"MappingWithDeposit";
+
+			let mut migrated_entries: Weight = 0;
+
+			// Move the mappings to the new storage location
+			// https://crates.parity.io/frame_support/storage/migration/fn.storage_key_iter.html
+			for (author_id, account_id) in frame_support::storage::migration::storage_key_iter::<
+				T::AuthorId,
+				RegistrationInfo<T::AccountId, BalanceOf<T>>,
+				Twox64Concat,
+			>(pallet_prefix, old_storage_item_prefix)
+			{
+				Mapping::<T>::insert(author_id, account_id);
+				migrated_entries += 1;
+			}
+
+			// Now remove the old storage
+			// https://crates.parity.io/frame_support/storage/migration/fn.remove_storage_prefix.html
+			frame_support::storage::migration::remove_storage_prefix(
+				pallet_prefix,
+				old_storage_item_prefix,
+				&[],
+			);
+
+			// Return the weight used.
+			let db_weights = T::DbWeight::get();
+			migrated_entries
+				.saturating_add(2 * db_weights.write)
+				.saturating_add(db_weights.read)
```

Comment:
Shouldn't the returned weight = `migrated_entries * cost_of_read + migrated_entries * cost_of_write + cost_of_read`? In this code, it is currently `migrated_entries + (2 * cost_of_write) + cost_of_read`

Reply:
because `migrated_entries` is the number of migrated entries, not the cost of migrating each entry

Reply:
Yeah, I had that very wrong. Thanks.